### ISSUE TITLE
Incompatibility with Smarter Trains :(

### ISSUE THREAD

**Issue by sgtsquiggs** [FKuerten/factorio-tree-growth-natural-growth#1]
Cross-posting this issue over at [Smarter Trains](https://github.com/Choumiko/SmartTrains)

```
Error while running event tree-growth-natural-growth::on_tick (ID 0)
Error while running event SmartTrains::Custom event (ID 71)
__SmartTrains__/control.lua:1232: attempt to index field 'entity' (a nil value)
stack traceback:
	__tree-growth-natural-growth__/control.lua:180: in function 'growTree'
	__tree-growth-natural-growth__/control.lua:194: in function <__tree-growth-natural-growth__/control.lua:184>
```

Running Factorio v0.15.12, Smarter Trains v2.0.3, Tree Growth v2.0.0

**Comment by sgtsquiggs** [comment]
Actually seems to be a problem with other mods as well:

```
Error while running event tree-growth-natural-growth::on_tick (ID 0)
Error while running event VehicleSnap::VehicleSnap-toggle (ID 71)
__VehicleSnap__/control.lua:8: attempt to index field '?' (a nil value)
stack traceback:
	__tree-growth-natural-growth__/control.lua:180: in function 'growTree'
	__tree-growth-natural-growth__/control.lua:194: in function <__tree-growth-natural-growth__/control.lua:184>
```

**Comment by FKuerten** [comment]
I installed Smarter Trains but there is no immediate error. Please explain how to reproduce this.
Edit: I can now reproduce this, but only if all three mods are present.

**Comment by FKuerten** [comment]
Can reproduce with Vehiclesnap, I will have a look.

**Comment by FKuerten** [comment]
Probably tgng uses a wrong event id, probably due to incorrect caching of the event ids. I should be able to fix this. This should also affect tgne and of course the real culprit is tg-core.

**Comment by FKuerten** [comment]
This has been fixed in 2.0.1. You will need to update tree-growth-core, tree-growh-natural-expansion and this one.
